x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -409,30 +559,27 @@ protected void doSaveState(IndexerState indexerState, Map<String, Object> positi
             }
 
             final DataFrameTransformState state = new DataFrameTransformState(
-                taskState.get(),
+                transformTask.taskState.get(),
                 indexerState,
                 getPosition(),
-                currentCheckpoint.get(),
-                stateReason.get());
-            logger.info("Updating persistent state of transform [" + transform.getId() + "] to [" + state.toString() + "]");
+                transformTask.currentCheckpoint.get(),
+                transformTask. stateReason.get());
+            logger.info("Updating persistent state of transform [" + transformConfig.getId() + "] to [" + state.toString() + "]");
 
             // Persisting stats when we call `doSaveState` should be ok as we only call it on a state transition and
             // only every-so-often when doing the bulk indexing calls.  See AsyncTwoPhaseIndexer#onBulkResponse for current periodicity
             ActionListener<PersistentTasksCustomMetaData.PersistentTask<?>> updateClusterStateListener = ActionListener.wrap(
                 task -> {
-                    // Make a copy of the previousStats so that they are not constantly updated when `merge` is called
-                    DataFrameIndexerTransformStats tempStats = new DataFrameIndexerTransformStats(previousStats).merge(getStats());
```

Comment:
There is no longer a need to keep track of previous stats and merge them as the stats dependency is given to the indexer constructor now and increments can continue as normal.